soc/riscv/riscv-ite/common/soc_dt.h

```diff
@@ -41,4 +41,120 @@
 		     inst)					\
 	}
 
+/*
+ * For it8xxx2 wake-up controller (WUC)
+ */
+#define IT8XXX2_DEV_WUC(idx, inst)					\
+	DEVICE_DT_GET(DT_PHANDLE(IT8XXX2_DT_INST_WUCCTRL_0(inst, idx), wucs))
+#define IT8XXX2_DEV_WUC_MASK(idx, inst)					\
+	DT_PHA(IT8XXX2_DT_INST_WUCCTRL_0(inst, idx), wucs, mask)
+
+/**
+ * @brief For it8xxx2, get a node identifier for a phandle in a wucctrl
+ *        property by index
+ *
+ * Example devicetree fragment:
+ *
+ *     n: node {
+ *             wucctrl-0 = <&foo &bar>;
+ *             wucctrl-1 = <&baz &blub>;
+ *     }
+ *
+ * Example usage:
+ *
+ *     IT8XXX2_DT_WUCCTRL_BY_IDX(DT_NODELABEL(n), 0, 1) // DT_NODELABEL(bar)
+ *     IT8XXX2_DT_WUCCTRL_BY_IDX(DT_NODELABEL(n), 1, 0) // DT_NODELABEL(baz)
+ *
+ * @param node_id node with a wucctrl-'pc_idx' property
+ * @param pc_idx index of the wucctrl property itself
+ * @param idx index into the value of the wucctrl property
+ * @return node identifier for the phandle at index 'idx' in 'wucctrl-'pc_idx''
+ */
+#define IT8XXX2_DT_WUCCTRL_BY_IDX(node_id, pc_idx, idx)			\
+	DT_CAT6(node_id, _P_wucctrl_, pc_idx, _IDX_, idx, _PH)
```

Comment:
Sorry, but this is an abstraction violation in multiple ways:

- The `DT_CAT<n>` macros are for internal use only within devicetree.h and its sub-headers; they are not public API. That is why they are not used anywhere outside of those headers right now. See `@internal` comment here: https://github.com/zephyrproject-rtos/zephyr/blob/ae8745df6ff5ad0b15b853a3ff271bfa2d26486e/include/devicetree.h#L3029.
- This is using the generated devicetree macros outside of devicetree.h and its sub-headers; the generated macros are private to those headers, and should not be used outside of them. See https://docs.zephyrproject.org/latest/guides/dts/api-usage.html#generated-macros, in particular "they should be considered an implementation detail"

We need to figure out another way to do this that preserves API boundaries.

This may take some time as we currently have special-cased `pinctrl-<N>` in gen_defines.py, so if you really need to do it this way with `wucctrl-<N>`, then we will need to figure out a way to generalize this special case.

Similar comments apply elsewhere.

Reply:
Done